### ISSUE TITLE
Failure when mocking functions that are run in parallel

### ISSUE THREAD

**Issue by filipesabella** [marick/Midje#209]
This test:

    (ns test
      (:require [midje.sweet :refer :all]))

    (defn my-func [x]
      (+ x 2))

    (defn my-func-caller [data]
      (pmap my-func data))

    (defn do-something []
      (my-func-caller [1 2 3]))

    (fact "Test some pcalls"
      (do-something) => [2 4 6]
      (provided
        (my-func 1) => 2
        (my-func 2) => 4
        (my-func 3) => 6))

    (fact "Test some pcalls 2"
      (do-something) => [2 4 6]
      (provided
        (my-func 1) => 2
        (my-func 2) => 4
        (my-func 3) => 6))

    (fact "Test some pcalls 3"
      (do-something) => [2 4 6]
      (provided
        (my-func 1) => 2
        (my-func 2) => 4
        (my-func 3) => 6))

Leads to this error:
 
    Actual: java.util.concurrent.ExecutionException: java.lang.Error: You seem to have created a prerequisite for
    #'test/my-func that interferes with that function's use in Midje's
    own code. To fix, define a function of your own that uses my-func, then
    describe that function in a provided clause. For example, instead of this:
      (provided (every? even? ..xs..) => true)
    do this:
      (def all-even? (partial every? even?))
      ;; ...
      (provided (all-even? ..xs..) => true)

It only breaks when there's more than one fact.
Changing from "pmap" to "map" on line 8 also fixes it.

Not sure how to proceed here.

**Comment by marick** [comment]
In the current implementation, parallelism is just out of scope.
